### PR TITLE
feat(cli): Automatically send influx-trace-id, and improve help text

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:
While analyzing customer performance in https://github.com/influxdata/EAR/issues/4236, I wanted to see tracing spans for a particular query run from the command line

I eventually figured out how to do it, but it requires changing the default tracing header name from `uber-tracing-id` to `influx-tracing-id` . See [slack conversation](https://influxdata.slack.com/archives/C04K1QRSDFG/p1684515393407249?thread_ts=1684510356.996419&cid=C04K1QRSDFG) for more details. 

# Changes

TLDR is you can now do:
```shell
influxdb_iox query namespace 'select count(*) from table'  where time >= now()' --gen-trace-id
```

And this will send spans from IOx to jaeger. It is influx cloud specific, but on the other hand `uber-trace-id` seems relatively specific too. 

I also improved the help text and added some examples of adding custom headers

### CODE REVIEW

File: influxdb_iox/src/main.rs [hunk 1]
```diff
@@ -384,13 +376,47 @@ fn main() -> Result<(), std::io::Error> {
...
 
-// Generates a compatible header values for a jaeger trace context header.
-fn gen_trace_id() -> String {
+/// configures tracing headers, so the remote server will sends
+/// tracing spans to Jaeger, if configured to do so.
+fn configure_tracing(mut builder: Builder, trace_id_header: &str) -> Builder {
+    let (trace_id, header_value) = gen_trace_id();
+    builder = set_trace_header(builder, &trace_id_header, &header_value);
+
+    // if the header is the default value (uber-trace-id), also set
```
Comment (alamb) [inline influxdb_iox/src/main.rs:385]: behavior change #1 is that now we also send the trace id in the `influx-trace-id` header as well as the `uber-trace-id` header

File: influxdb_iox/src/main.rs [hunk 2]
```diff
@@ -384,13 +376,47 @@ fn main() -> Result<(), std::io::Error> {
...
+    // the influx cloud
+    if trace_id_header == trace_exporters::DEFAULT_JAEGER_TRACE_CONTEXT_HEADER_NAME {
+        builder = set_trace_header(
+            builder,
+            trace_exporters::DEFAULT_INFLUX_TRACE_CONTEXT_HEADER_NAME,
+            &header_value,
+        );
+    }
+
+    println!("Trace ID set to {trace_id}");
```
Comment (alamb) [inline influxdb_iox/src/main.rs:396]: behavior change #2 is that we now print out only the trace id (that can be fed directly to jaeger) rather than the entire header value (which is still logged via `debug!` if you want to see it)

File: influxdb_iox/src/main.rs [hunk 3]
```diff
@@ -127,7 +127,9 @@ struct Config {
 
     /// Additional headers to add to CLI requests
     ///
-    /// Values should be key value pairs separated by ':'
+    /// Values should be key value pairs separated by ':'. For example:
+    /// `foo:bar` or
+    /// `influx-trace-id:"f52000bb08c9520:1112223334445:0:1`
```
Comment (appletreeisyellow) [inline influxdb_iox/src/main.rs:132]:
```suggestion
    /// `influx-trace-id:"f52000bb08c9520:1112223334445:0:1"`
```

File: influxdb_iox/src/main.rs [hunk 4]
```diff
@@ -140,7 +142,7 @@ struct Config {
     )]
     rpc_timeout: Duration,
 
-    /// Trace ID header.
+    /// HTTP header name for the Trace ID header
     ///
     /// See `--gen-trace-id` to trigger header generation.
```
Comment (appletreeisyellow) [inline influxdb_iox/src/main.rs:147]: 👍

File: influxdb_iox/src/main.rs [hunk 5]
```diff
@@ -384,13 +376,47 @@ fn main() -> Result<(), std::io::Error> {
...
+/// returns (trace_id, header_value)
+fn gen_trace_id() -> (String, String) {
     let now = SystemProvider::new().now();
     let mut hasher = DefaultHasher::new();
     now.timestamp_nanos().hash(&mut hasher);
 
-    format!("{:x}:1112223334445:0:1", hasher.finish())
+    let trace_id = format!("{:x}", hasher.finish());
+    let header_value = format!("{trace_id}:1112223334445:0:1");
+    (trace_id, header_value)
```
Comment (appletreeisyellow) [inline influxdb_iox/src/main.rs:419]: Just curious what does `1112223334445:0:1` represent?

Comment (alamb) [reply influxdb_iox/src/main.rs:419]: I believe it is the so-called "Jaeger Propagation Format" 

https://www.jaegertracing.io/docs/1.22/client-libraries/#value

So `{trace-id}:{span-id}:{parent-span-id}:{flags}`

But I will confess to simply cargo culting it. Maybe @tustvold  knows more

Comment (alamb) [reply influxdb_iox/src/main.rs:419]: So `1112223334445` is some (I think hard coded) span id, and the flag means "sample it!"

File: influxdb_iox/src/main.rs [hunk 6]
```diff
@@ -151,11 +153,12 @@ struct Config {
     )]
     trace_id_header: String,
```
Comment (crepererum) [inline influxdb_iox/src/main.rs:154]:
```suggestion
    trace_id_headers: Vec<String>,
```

and add a proper default (e.g. two headers) instead of silently hacking in another header if the default is set.

**Review by crepererum — CHANGES_REQUESTED** [review]
I think this should go through a proper clap config instead of some undocumented hack.

File: influxdb_iox/src/main.rs (hunk 6 above)
Comment (alamb) [reply influxdb_iox/src/main.rs:154]: Good idea -- will do so

Comment (alamb) [reply influxdb_iox/src/main.rs]: in 36bf37b01

Now the help text looks like:

```
      --trace-id-header <TRACE_ID_HEADER>
          HTTP header names sent with Trace ID information
          
          See `--gen-trace-id` to trigger automatic header generation.
          
          [default: uber-trace-id influx-trace-id]
```

And it still works great:

```shell
influxdb_iox -vv --gen-trace-id query NS '<REDACTED>'
2023-05-24T18:14:16.784630Z DEBUG influxdb_iox: Setting trace header name="uber-trace-id" value="c9dda0ba17269e15:1112223334445:0:1"
2023-05-24T18:14:16.784689Z DEBUG influxdb_iox: Setting trace header name="influx-trace-id" value="c9dda0ba17269e15:1112223334445:0:1"
Trace ID set to c9dda0ba17269e15
2023-05-24T18:14:16.785067Z DEBUG hyper::client::connect::http: connecting to 127.0.0.1:8082
2023-05-24T18:14:16.785685Z DEBUG hyper::client::connect::http: connected to 127.0.0.1:8082
2023-05-24T18:14:16.787333Z DEBUG tower::buffer::worker: service.ready=true processing request
... REDACTED ...
Compilation finished at Wed May 24 14:14:18
```

**Review by crepererum — APPROVED** [review]
Thank you.
